prereq_tools/base.py

```diff
@@ -498,13 +498,13 @@ def get(self, subdir, **kw):
         """Downloads and extracts sources from a url into subdir"""
         basename = os.path.basename(self.url)
 
-        if os.path.exists(basename) and os.path.exists(subdir):
+        if os.path.exists(subdir):
             # assume that nothing has changed
             return
 
-        commands = ['rm -rf %s' % subdir]
-        if not os.path.exists(basename):
-            commands.append('curl -L -O %s' % self.url)
+        # Always removes the tar file to avoid using garbage
+        commands = ['rm -rf %s %s' % (subdir, basename),
```

Comment:
My only concern with this is that we could trigger somebody's abuse thresholds by continually downloading the same thing for every build.

An alternative is to use something like `file` (or better, a Python equivalent module rather than a shell-out) on the file to see if it's at least the correct type and then remove only if it's not.

Reply:
For local builds, it should skip the download because hwloc exists.  For jenkins builds, do we not attempt to download every time already?